scripts/generate-doc.sh

```diff
@@ -70,6 +70,8 @@ fi
 # ensure preview resource and data sources are also included during generation
 export MONGODB_ATLAS_ENABLE_PREVIEW="true" 
 
+trap 'rm -R docs-out/' EXIT # temp dir cleanup when script exits
+
```

Comment:
remove docs-out temp folder at the end of script execution even if the scripts fails in the middle